lib/volumes.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright (c) 2017, Joyent, Inc.
+ */
+
+var assert = require('assert-plus');
+
+function throwInvalidSize(size) {
+    assert.string(size, 'size');
+
+    throw new Error('size "' + size + '" is not a valid volume size');
+}
+
+/*
+ * Returns the number of MiBs (Mebibytes) represented by the string "size". That
+ * string can have different format suffixes, such as "100GB", "100G", etc.
+ * If "size" is not a valid size string, an error is thrown.
+ */
+function parseVolumeSize(size) {
+    assert.optionalString(size, 'size');
+
+    var MIBS_IN_GB = 1024;
+
+    var MULTIPLIERS_TABLE = {
+        g: MIBS_IN_GB,
+        GB: MIBS_IN_GB,
+        m: 1,
+        MB: 1
+    };
+
+    var multiplierSymbol, multiplier;
+    var baseValue;
+
+    if (size === undefined) {
+        return undefined;
+    }
+
+    var matches = size.match(/(\d+)(g|m|G|M|gb|mb|GB|MB)/);
```

Comment:
`STRING.match(/.../)` only anchors at the start.

    > '8g'.match(/(\d+)(g|m|G|M|gb|mb|GB|MB)/)
    [ '8g', '8', 'g', index: 0, input: '8g' ]
    > '8g'.match(/(\d+)(g|m|G|M|gb|mb|GB|MB)/)
    > '8gasdf'.match(/(\d+)(g|m|G|M|gb|mb|GB|MB)/)
    [ '8g', '8', 'g', index: 0, input: '8gasdf' ]